File: ZLib/app/src/main/java/com/example/zlib/data/Book.kt
package com.example.zlib.data
import com.google.gson.annotations.SerializedName

data class Book(
    @SerializedName("id") val id: Int,
    @SerializedName("title") val title: String,
    @SerializedName("author") val author: String,
    @SerializedName("pageCount") val pageCount: Int,
    @SerializedName("currentPage") val currentPage: Int = 0,
    @SerializedName("isbn") val isbn: String,
    @SerializedName("status") val statusInt: Int = 1,
    @SerializedName("description") val description: String? = null
){
    val status: BookStatus
        get() = BookStatus.fromInt(statusInt)
}

File: Backend/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using Backend.Enums;

namespace Backend.Models
{
    public class Book
    {
        [Key]
        public int Id { get; set; }

        public required string Title { get; set; }

        public required string Author { get; set; }
        public string? Isbn { get; set; }

        public string? Description { get; set; }
        public int PageCount { get; set; }

        public int CurrentPage { get; set; } = 0;
        public BookStatus Status { get; set; } = BookStatus.Unread;
        
        public int? Rating { get; set; }

        public string? ImagePath { get; set; }
        
    }
}